RevenueCatUI/CustomerCenter/Views/PromotionalOfferView.swift

```diff
@@ -94,6 +100,33 @@ struct PromotionalOfferView: View {
     ) {
         self.onDismissPromotionalOfferView(action) // Forward results to parent view
     }
+
+    private struct AppIconView: View {
+        var body: some View {
+            if let appIcon = AppIconHelper.getAppIcon() {
+                Image(uiImage: appIcon)
+                    .resizable()
+                    .aspectRatio(contentMode: .fit)
+                    .frame(width: 100, height: 100)
+                    .clipShape(RoundedRectangle(cornerRadius: 20, style: .continuous))
+                    .shadow(radius: 10)
+            } else {
+                EmptyView()
+            }
+        }
+    }
+
+    private enum AppIconHelper {
+        static func getAppIcon() -> UIImage? {
+            guard let iconsDictionary = Bundle.main.infoDictionary?["CFBundleIcons"] as? [String: Any],
+                  let primaryIcons = iconsDictionary["CFBundlePrimaryIcon"] as? [String: Any],
+                  let iconFiles = primaryIcons["CFBundleIconFiles"] as? [String],
+                  let lastIcon = iconFiles.last else {
+                return nil
+            }
+            return UIImage(named: lastIcon)
```

Comment:
just fetch the app icon, seems to work well in my testing

Reply:
🤔 looks a bit blurry actually, maybe I can do something to ensure we're always getting the highest-res version